cilium-dbg/cmd/encrypt_status.go

```diff
@@ -57,71 +58,93 @@ func init() {
 }
 
 func getEncryptionStatus() (models.EncryptionStatus, error) {
+	var result models.EncryptionStatus
+
 	params := daemon.NewGetHealthzParamsWithTimeout(timeout)
 	params.SetBrief(&brief)
+
 	resp, err := client.Daemon.GetHealthz(params)
 	if err != nil {
-		return models.EncryptionStatus{}, err
+		result.Msg = err.Error()
+	} else {
+		result.Mode = resp.Payload.Encryption.Mode
 	}
 
```

Comment:
```suggestion
	enc := resp.Payload.Encryption
	switch enc.Mode {
	case models.EncryptionStatusModeIPsec:
		return dumpIPsecStatus()
	case models.EncryptionStatusModeWireguard:
		return dumpWireGuardStatusFromAgent(enc), nil
	default:
		// here let's not return err in case cilium_wg0 does not exist
		result.Wireguard, err = dumpWireGuardStatus()
		if err != nil {
			return result, err
		}
		result.Ipsec, err = dumpIPsecStatus()
		if err != nil {
			return result, err
		}
	}
	return result, nil 
```

What's your opinion on this?
* if we received a reply from agent:
   * If Wireguard, we already have data to display, so use the response payload
   * If IPSec, skip wireguard
* in case the agent did not respond, try both and don't fail at obvious errors (eg if cilium_wg0 does not exist we should still continue and check IPSec)

Reply:
i believe there is value in retrieving everything regardless, see https://github.com/cilium/cilium/pull/41827#discussion_r2378287723

Reply:
Understood, I see... but I'd not ignore the agent reply.
That would be the expected outcome. We can still do our lookup here and if it doesn't match then we report the diff.

Reply:
the agent reply isn't ignored, really - the difference is that the reply is only used to populate the status field (the one that says ipsec or wireguard)

Reply:
1. `cilium debuginfo` calls WGAgent.Status(true) and gets peers info
2. `cilium encrypt status -o json` calls WGAgent.Status(false), no peer info, but it at least returns the list of interfaces (cilium_wg0)

You would basically use the agent reply to check only if IPSec/WG, and then populate other fields by running commands anyway, right? If so, WGAgent.Status() would be useful also for (1).

That's just to make sure I'm getting the point, thanks for the comprehension 🙏🏼

Reply:
the big difference between the debuginfo command and the `cilium encrypt status` is that the debuginfo call contacts the agent and the Status() method relies on agent state (like a.wgConn, a.wgConfig, and so on) - which aren't available at the cilium-dbg runtime - unless the agent can be contacted successfully.

so you are basically saying that (at least for wireguard), if we are able to contact the agent in the cilium-dbg command, use the agent to retrieve the wg state from the kernel, else fallback to using the wireguard library to query for the state. did i get it right?

Reply:
What I'm saying is that IF the agent replies, we could double check that the reply (list of interfaces) is at least included in the state you're retrieving. In case they're mismatching, I'd add two lines in the Msg.
What do you think?

Reply:
it makes sense!